daisy/step_resize_disks.go

```diff
@@ -0,0 +1,100 @@
+//  Copyright 2017 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package daisy
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"google.golang.org/api/compute/v1"
+)
+
+// ResizeDisks is a Daisy ResizeDisks workflow step.
+type ResizeDisks []*ResizeDisk
+
+// ResizeDisk is used to resize a GCE disk.
+type ResizeDisk struct {
+	compute.DisksResizeRequest
+	// Name of the disk to be resized
+	Name          string
+	Project, Zone string
+}
+
+func (r *ResizeDisks) populate(ctx context.Context, s *Step) dErr {
+	for _, rd := range *r {
+		if diskURLRgx.MatchString(rd.Name) {
+			rd.Name = extendPartialURL(rd.Name, s.w.Project)
+		}
+		if rd.Project == "" {
+			rd.Project = s.w.Project
+		}
+		if rd.Zone == "" {
+			rd.Zone = s.w.Zone
+		}
+	}
+	return nil
+}
+
+func (r *ResizeDisks) validate(ctx context.Context, s *Step) dErr {
+	var errs dErr
+	for _, rd := range *r {
+		dr, err := s.w.disks.regUse(rd.Name, s)
+		if dr == nil {
+			// Return now, the rest of this function can't be run without dr.
+			return addErrs(errs, errf("cannot resize disk: %v", err))
+		}
+		// Reference the actual name of the disk
+		rd.Name = dr.RealName
+
+		pre := fmt.Sprintf("cannot resize disk %q", rd.Name)
+		if rd.SizeGb <= 0 {
```

Comment:
The API wont allow you to decrease the size of the disk either. We should add a check for that as well here. Feel free to just add a TODO comment for this.

Reply:
can we check the current disk size on `validate`? I'm afraid the disk information would not be there (if you give an image, you don't know how big it would be at that point).

I'm considering doing that on `run`, ok?

Reply:
Doing it in run is a bit pointless as at that point you can just make the API call. Having this in populate would be a bit tough without recording info on disk sizes in the disk registry